app-header/app-header.html

```diff
@@ -0,0 +1,223 @@
+      height: 5px;
+      pointer-events: none;
+      opacity: 0;
+      box-shadow: inset 0px 5px 6px -3px rgba(0, 0, 0, 0.4);
+      -webkit-transition: opacity 0.4s;
+      transition: opacity 0.4s;
+      /* fix bug in chrome */
+      border-top: 1px solid transparent;
+    }
+
+    :host([shadow])::after {
+      opacity: 1; 
+    }
+
+  </style>
+
+  <template>
+
+    <content></content>
+
+  </template>
+
+  <script>
+
+    Polymer({
+
+      is: 'app-header',
+
+      properties: {
+
+        /**
+         * If true, the header's height will condense as the user scrolls down
+         * from the top of the content area.
+         */
+        condenses: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * If true, the header is fixed to the top and never moves away.
+         */
+        fixed: {
+          type: Boolean,
+          value: false,
+          observer: '_fixedChanged'
+        },
+
+        /**
+         * If true, the condensed header is always shown and does not move away.
+         */
+        alwaysVisible: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * If true, a shadow will appear when the user is not at the top of the content area.
+         */
+        waterfall: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * If true, a shadow will appear when the user is not at the top of the content area.
+         */
+        shadow: {
+          type: Boolean,
+          readOnly: true,
+          reflectToAttribute: true,
+          value: false
+        }
+
+      },
+
+      behaviors: [
+        Polymer.CustomScrollerBehavior
+      ],
+
+      /**
+       * A cached offsetHeight of the current element
+       */
+      _height: 0,
+
+      /**
+       * The top most app-toolbar
+       */
+      _topbar: null,
+
+      /**
+       * The distance in pixels the header will be translated to when scrolling.
+       */
+      _deltaHeight: 0,
+
+      get _clampedScrollTop() {
+        return Math.max(0,
+            this.alwaysVisible ? Math.min(this._scrollTop, this._deltaHeight) : this._scrollTop);
+      },
+
+      get _headerDy() {
+        return this.alwaysVisible ? this._deltaHeight : this._height;
+      },
+
+      created: function() {
+        this._oldY = 0;
+        this._oldTop = 0;
+      },
+
+      attached: function() {
+        this.async(this._setup, 1);
+      },
+
+      _setup: function() {
+        this._height = this.offsetHeight;
+        this._topbar = Polymer.dom(this).querySelectorAll('app-toolbar')[0];
+        this._deltaHeight = this._topbar ? this._height - this._topbar.offsetHeight : 0;
+      },
+
+      _scrollHandler: function() {
+        this._updateHeader(this._clampedScrollTop);
+      },
+
+      /**
+       * Fired when the header is translated.
+       *
+       * @event app-header-transform
+       */
+      _updateHeader: function(scrollTop) {
```

Comment:
This looks like docs for `_updateHeader` function but is really for event.  Could you move the doc right after `is: app-header`?